types/src/transaction/mod.rs

```diff
@@ -736,6 +741,11 @@ impl SignedTransaction {
         Ok(SignatureCheckedTransaction(self))
     }
 
+    pub fn verify_signature(&self) -> Result<()> {
```

Comment:
it's confusing to have this and `check_signature`, `SignatureCheckedTransaction` and `SignatureVerifiedTransaction` why do we need to keep the invalid case?

Reply:
@zekun000 - We need to keep the invalid case as we want to notify the mempool/state sync to remove these txns in the commit path.

Reply:
I think it's better to guarantee the block only has valid signature txns (like check in quorum store) but I guess that's for later

Reply:
Yes, that is true.. it would be much cleaner to pass only valid sig txn to the executor.. let me add a TODO here and we can follow up on that.